src/components/tabs/tabs.component.js

```diff
@@ -30,57 +29,30 @@ const Tabs = ({
   selectedTabId,
   renderHiddenTabs = true,
   position = "top",
-  setLocation = true,
   extendedLine = true,
   size,
   borders = "off",
   variant = "default",
   validationStatusOverride,
   ...rest
 }) => {
-  const firstRender = useRef(true);
   const tabRefs = useRef([]);
   const previousSelectedTabId = useRef(selectedTabId);
   const [selectedTabIdState, setSelectedTabIdState] = useState();
   const sidebarContext = useContext(SidebarContext);
   const [tabsErrors, setTabsErrors] = useState({});
   const [tabsWarnings, setTabsWarnings] = useState({});
   const [tabsInfos, setTabsInfos] = useState({});
-  const _window = Browser.getWindow();
-  const isInSidebar = !!(sidebarContext && sidebarContext.isInSidebar);
+  const isInSidebar = !!sidebarContext?.isInSidebar;
 
   useLayoutEffect(() => {
-    if (firstRender.current) {
-      firstRender.current = false;
-      let selectedTab;
-      if (selectedTabId) {
-        selectedTab = selectedTabId;
-      } else {
-        const hash = _window.location.hash.substring(1);
-
-        if (Array.isArray(children)) {
-          const filteredChildren = children.filter((child) => child);
-          let useHash = false;
-
-          if (hash) {
-            for (const index in filteredChildren) {
-              const child = filteredChildren[index];
-
-              if (child.props.tabId === hash) {
-                useHash = true;
-                break;
-              }
-            }
-          }
-
-          selectedTab = useHash ? hash : filteredChildren[0].props.tabId;
-        } else {
-          selectedTab = children.props.tabId;
-        }
-      }
-      setSelectedTabIdState(selectedTab);
-    }
-  }, [_window.location.hash, children, firstRender, selectedTabId]);
+    const selectedTab =
+      selectedTabId ||
+      React.Children.toArray(children).filter((child) => child)[0].props.tabId;
```

Comment:
What's the point of the `.filter((child) => child)` bit here? This could just be `React.Children.toArray(children)[0].props.tabId;` I think

Reply:
Yeah I added it to account for conditional rendering of the `Tab` `children` but I think you're right and it's overkill so I'll update it